src/calendar/Calendar.ts

```diff
@@ -0,0 +1,391 @@
+import { WidgetBase, onPropertiesChanged } from '@dojo/widget-core/WidgetBase';
+import { ThemeableMixin, ThemeableProperties, theme } from '@dojo/widget-core/mixins/Themeable';
+import WidgetRegistry from '@dojo/widget-core/WidgetRegistry';
+import { v, w } from '@dojo/widget-core/d';
+import { DNode, PropertiesChangeEvent } from '@dojo/widget-core/interfaces';
+import uuid from '@dojo/core/uuid';
+import { includes } from '@dojo/shim/array';
+import { Keys } from '../common/util';
+import MonthPicker, { CalendarMessages } from './MonthPicker';
+import CalendarCell, { CalendarCellProperties } from './CalendarCell';
+import * as css from './styles/calendar.m.css';
+
+/**
+ * @type CalendarProperties
+ *
+ * Properties that can be set on a Calendar component
+ *
+ * @property customDateCell    Custom widget constructor for the date cell. Should use CalendarCell as a base.
+ * @property labels            Customize or internationalize accessible text for the Calendar widget
+ * @property month             Set the currently displayed month, 0-based
+ * @property monthNames        Customize or internationalize full month names and abbreviations
+ * @property selectedDate      The currently selected date
+ * @property weekdayNames      Customize or internationalize weekday names and abbreviations
+ * @property year              Set the currently displayed year
+ * @property renderMonthLabel  Format the displayed current month and year
+ * @property renderWeekdayCell Format the weekday column headers
+ * @property onMonthChange     Function called when the month changes
+ * @property onYearChange      Function called when the year changes
+ * @property onDateSelect      Function called when the user selects a date
+ */
+export interface CalendarProperties extends ThemeableProperties {
+	customDateCell?: any;
```

Comment:
Well I would say we'd probably want to type them correctly everywhere, I think it would be `Constructor<Cell>` or `Constructor<WidgetBase>`? No? Assuming it accepts a `WidgetConstructor`?